File: API-NOVA/src/main/java/br/projeto/petshop/model/Status.java
package br.projeto.petshop.model;

import com.fasterxml.jackson.annotation.JsonFormat;

@JsonFormat(shape = JsonFormat.Shape.OBJECT)
public enum Status {
    AGUARDANDO_PAGAMENTO(1, "Aguardando pagamento"),
    PAGAMENTO_APROVADO(2, "Pagamento aprovado"),
    PAGAMENTO_REJEITADO(3, "Pagamento rejeitado"),
    EM_PROCESSAMENTO(4, "Em processamento"),
    PRONTO_PARA_ENVIO(5, "Pronto para envio"),
    ENVIADO(6, "Enviado"),
    EM_TRANSPORTE(7, "Em transporte"),
    ENTREGUE(8, "Entregue"),
    CONCLUIDO(9, "Concluido"),
    CANCELADO(10, "Cancelado"),
    DEVOLUCAO_SOLICITADA(11, "Devolução solicitada"),
    EM_DEVOLUCAO(12, "Em devolução"),
    DEVOLVIDO(13, "Devolvido"),
    REEMBOLSADO(14, "Reembolsado");

    private final Integer id;
    private final String label;
    private Status(Integer id, String label) {
        this.id = id;
        this.label = label;
    }
    public Integer getId() {
        return id;
    }
    public String getLabel() {
        return label;
    }

    public static Status valueOf(Integer id) throws IllegalArgumentException {
        if (id == null) {
            return null;
        }
        for (Status status : Status.values()) {
            if (status.getId().equals(id)) {
                return status;
            }
        }
        throw new IllegalArgumentException("ID invalido para Status: " + id);
    }

    
}

File: angular-pet-shop/src/app/models/status.ts
export enum Status {
    AGUARDANDO_PAGAMENTO = 1,
    PAGAMENTO_APROVADO = 2,
    PAGAMENTO_REJEITADO = 3,
    EM_PROCESSAMENTO = 4,
    PRONTO_PARA_ENVIO = 5,
    ENVIADO = 6,
    EM_TRANSPORTE = 7,
    ENTREGUE = 8,
    CONCLUIDO = 9,
    CANCELADO = 10,
    DEVOLUCAO_SOLICITADA = 11,
    EM_DEVOLUCAO = 12,
    DEVOLVIDO = 13,
    REEMBOLSADO = 14
}

export const StatusLabel = {
    [Status.AGUARDANDO_PAGAMENTO]: 'Aguardando pagamento',
    [Status.PAGAMENTO_APROVADO]: 'Pagamento aprovado',
    [Status.PAGAMENTO_REJEITADO]: 'Pagamento rejeitado',
    [Status.EM_PROCESSAMENTO]: 'Em processamento',
    [Status.PRONTO_PARA_ENVIO]: 'Pronto para envio',
    [Status.ENVIADO]: 'Enviado',
    [Status.EM_TRANSPORTE]: 'Em transporte',
    [Status.ENTREGUE]: 'Entregue',
    [Status.CONCLUIDO]: 'Concluído',
    [Status.CANCELADO]: 'Cancelado',
    [Status.DEVOLUCAO_SOLICITADA]: 'Devolução solicitada',
    [Status.EM_DEVOLUCAO]: 'Em devolução',
    [Status.DEVOLVIDO]: 'Devolvido',
    [Status.REEMBOLSADO]: 'Reembolsado'
};
